File: app/src/main/java/com/waygo/data/model/butler/ButlerResponse.java
package com.waygo.data.model.butler;

import android.support.annotation.NonNull;

import com.waygo.utils.result.ResultJ;

public abstract class ButlerResponse {

    @NonNull
    private final String mMessage;

    protected ButlerResponse(@NonNull final String message) {
        mMessage = message;
    }

    @NonNull
    public String getMessage() {
        return mMessage;
    }

    @NonNull
    protected static ResultJ<String> createMessage(@NonNull final String message) {
        return ResultJ.asResult(message)
                .filter(s -> !s.isEmpty(), __ -> "Sorry, didn't get that.");
    }
}

File: app/src/main/java/com/waygo/data/model/butler/ButlerResponse.kt
package com.waygo.data.model.butler

import com.waygo.utilskt.Result
import kotlin.platform.platformStatic

public abstract class ButlerResponse protected constructor(public val message: String) {
    companion object {

        protected platformStatic fun createMessage(message: String): Result<String> =
                Result.ofObj(message)
                    .filter( {!it.isEmpty()})  {"Sorry, didn't get that."}
    }
}
